src/loaders/utils.js

```diff
@@ -102,3 +104,75 @@ export function isBioformatsNoPadHeightVersion(software) {
   }
   return false;
 }
+
+/**
+ * Returns actual image stats for static imagery and an estimate via a downsampled version of image pyramids.
+ * This is helpful for generating histograms of your channel data, or scaling your sliders down to a reasonable range.
+ * @param {Object} args
+ * @param {Object} args.loader A valid loader object.
+ * @param {Array} args.loaderSelection Array of valid dimension selections
+ * @returns {Object} { means, dataRanges, standardDeviations, data, firstQuartiles, thirdQuartiles }, each arrays with entries for each channel based on your selection.
+ */
+export async function getChannelStats({ loader, loaderSelection }) {
+  const z = loader.isPyramid ? loader.numLevels - 1 : 0;
+  const rasters = await loader.getRaster({ z, loaderSelection });
+  const { data } = rasters;
+  const channelStats = {
+    means: [],
+    medians: [],
+    dataRanges: [],
+    standardDeviations: [],
+    data,
+    firstQuartiles: [],
+    thirdQuartiles: []
+  };
+  data.forEach(arr => {
+    let len = arr.length;
+    let min = Infinity;
+    let max = -Infinity;
+    let total = 0;
+    // Range (min/max).
+    // eslint-disable-next-line no-plusplus
+    while (len--) {
+      if (arr[len] < min) {
+        min = arr[len];
+      }
+      if (arr[len] > max) {
+        max = arr[len];
+      }
+      total += arr[len];
+    }
```

Comment:
any reason not to use `for ... of`?

```suggestion
    for (let val of arr) {
      if (val < min) min = val;
      if (val > max) max = val;
      total += val
    }
```

Reply:
Big performance gains to the while loop: https://jsbench.me/rtkawo35dg/1
<img width="1660" alt="Screen Shot 2020-06-01 at 11 49 29 AM" src="https://user-images.githubusercontent.com/43999641/83427014-0cccf200-a3fe-11ea-8eec-6693669d0eed.png">